docs/how-to/create_plugin.rst

```diff
@@ -0,0 +1,131 @@
+
+.. code:: py
+
+    class NewDumpPluginProperties(PluginProperties, frozen=True):
+        """The part properties used by the new dump plugin."""
+
+        # The human-friendly name of the plugin
+        plugin: Literal["new-dump"] = "new-dump"
+        # The "source" parameter is special - we'll talk about it more soon.
+        source: str
+
+Creating the plugin class
+~~~~~~~~~~~~~~~~~~~~~~~~~
+The plugin class itself contains the functionality of the plugin. Start by
+declaring the properties class:
+
+.. code:: py
+
+    class NewDumpPlugin(Plugin):
+        """Copy the content from the part source."""
+        
+        properties_class = NewDumpPluginProperties
+    
+Next, we'll have to start defining the special methods that make this plugin
+"work". Since this is such a simple plugin, many of these methods are empty:
+
+.. code:: py
+
+    @override
+    def get_build_snaps(self) -> set[str]:
+        """Return a set of required snaps to install in the build
+        environment."""
+        return set()
+
+    @override
+    def get_pull_commands(self) -> list[str]:
+        """Return a list of commands to retrieve dependencies during the
+        pull step."""
+        return []
+
+    @override
+    def get_build_packages(self) -> set[str]:
+        """Return a set of required packages to install in the build
+        environment."""
+        return set()
+
+    @override
+    def get_build_environment(self) -> dict[str, str]:
+        """Return a dictionary with the environment to use in the build
+        step."""
+        return {}
+
+The last method that we have to define, however, is where the actual build
+commands are defined. These are the exact commands executed by a subprocess
+during the build process:
+
+.. code:: py
+
+    @override
+    def get_build_commands(self) -> list[str]:
+        """Return a list of commands to run during the build step."""
+        install_dir = self._part_info.part_install_dir
+        return [f'cp --archive --link --no-dereference . "{install_dir}"']
+
+Notice we don't require a user-provided source directory/file - this is because
+we let the ``source`` parameter do the heavy-lifting. The source parameter is a
+mandatory parameter that specifies the working files of a part. It has a lot of
+specialised logic for different file types ranging from tar files to git
+repositories, so we can leverage that to get much stronger functionality out of
+our otherwise simple plugin.
+
+Final steps
+~~~~~~~~~~~
+Now that we have our very own plugin, the last step is to make the lifecycle
+manager aware of this plugin. Since we created our plugin directly in
+``craft-parts``, all that's needed is to add it to a dictionary in
+:file:`craft_parts/plugins/plugins.py`:
+
+.. code:: py
+
+    from .new_dump_plugin import NewDumpPlugin
+
+    # ...
+
+    _BUILTIN_PLUGINS: dict[str, PluginType] = {
+        # ...
+        "new-dump": NewDumpPlugin,
+    }
```

Comment:
I know the other how-tos in Craft Parts are still nascent, but have you considered placing some links to further tasks or references in a _Next steps_ section?